### PR TITLE
 refactor: use block.timestamp instead of block.number

### PR DESCRIPTION
**Pull request by xorsal**
# 🤖 Linear

Closes OPO-647

Tests pass using [this version](https://github.com/defi-wonderland/prophet-core/pull/46/) of prophet-core.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -38,7 +38,7 @@
     "package.json": "sort-package-json"
   },
   "dependencies": {
-    "@defi-wonderland/prophet-core": "0.0.0-0954a47f",
+    "@defi-wonderland/prophet-core": "github:defi-wonderland/prophet-core#de18a05",
```
Comment (gas1cent) [inline package.json:41]: Let's wait till https://github.com/defi-wonderland/prophet-core/pull/46 is merged, release a new canary of the core package and use it here.

File: solidity/test/unit/modules/dispute/BondedDisputeModule.t.sol [hunk 2]
```diff
@@ -259,7 +259,7 @@ contract BondedDisputeModule_Unit_DisputeResponse is BaseTest {
 
     // Expect the event
     vm.expectEmit(true, true, true, true, address(bondedDisputeModule));
-    emit ResponseDisputed(_requestId, _getId(mockResponse), _getId(mockDispute), mockDispute, block.number);
+    emit ResponseDisputed(_requestId, _getId(mockResponse), _getId(mockDispute), mockDispute, block.timestamp);
```
Comment (gas1cent) [inline solidity/test/unit/modules/dispute/BondedDisputeModule.t.sol:262]: Careful here, the event is [defined](https://github.com/defi-wonderland/prophet-core/blob/fc4e298ea4e19778e06870abfc6ced8ba8f1ef0c/solidity/interfaces/modules/dispute/IDisputeModule.sol#L12-L25) as:
```solidity
/**
  * @notice Emitted when a response is disputed
  * @param _responseId The id of the response disputed
  * @param _disputeId The id of the dispute
  * @param _dispute The dispute that is being created
  * @param _blockNumber The current block number
  */
event ResponseDisputed(
  bytes32 indexed _requestId,
  bytes32 indexed _responseId,
  bytes32 indexed _disputeId,
  IOracle.Dispute _dispute,
  uint256 _blockNumber
);
```

Comment (xorsal) [reply solidity/test/unit/modules/dispute/BondedDisputeModule.t.sol:262]: Good catch.
I blame my urge to update the events.

Comment (xorsal) [reply solidity/test/unit/modules/dispute/BondedDisputeModule.t.sol]: Misused param was removed in 5590bd0

**Review by gas1cent — COMMENTED** [review]
The code looks good to me, waiting for the core PR to be merged.

File: package.json (hunk 1 above)
Comment (gas1cent) [inline package.json:41]: [Latest canary](https://www.npmjs.com/package/@defi-wonderland/prophet-core?activeTab=versions)
```suggestion
    "@defi-wonderland/prophet-core": "0.0.0-4d28a0ec",
```

**Review by 0xJabberwock — APPROVED** [review]
LGTM!

**Review by gas1cent — APPROVED** [review]

**Pull request merged**